src/containerapp/azext_containerapp/containerapp_decorator.py

```diff
@@ -939,49 +969,32 @@ def post_process(self, r):
                 while r["properties"]["provisioningState"].lower() == "inprogress":
                     r = self.client.show(self.cmd, self.get_argument_resource_group_name(), self.get_argument_name())
                     time.sleep(1)
-                linker_client.linker.begin_create_or_update(resource_uri=r["id"],
-                                                            parameters=item["parameters"],
-                                                            linker_name=item["linker_name"]).result()
+
+                linker_create_or_update(
+                    linker_client=linker_client,
+                    r=r,
+                    parameters=item["parameters"][0],
+                    linker_name=item["linker_name"])
         return r
 
     def set_up_service_bindings(self):
         if self.get_argument_service_bindings() is not None:
-            linker_client = get_linker_client(self.cmd)
-
             service_connectors_def_list, service_bindings_def_list = parse_service_bindings(self.cmd, self.get_argument_service_bindings(), self.get_argument_resource_group_name(), self.get_argument_name())
+            # Case: Kafka on Confluent Cloud. Update list with 2-parameter connectors
+            service_connectors_def_list = update_connectors_with_two_parameters(service_connectors_def_list)
+
+            check_bindings_and_raise_error(self.cmd, service_connectors_def_list, service_bindings_def_list, self.get_argument_resource_group_name(), self.get_argument_name())
+            # Only set these for post_process if no errors were raised
             self.set_argument_service_connectors_def_list(service_connectors_def_list)
-            service_bindings_used_map = {update_item["name"]: False for update_item in service_bindings_def_list}
+            self.set_argument_service_bindings_def_list(service_bindings_def_list)
 
             safe_set(self.new_containerapp, "properties", "template", "serviceBinds", value=self.containerapp_def["properties"]["template"]["serviceBinds"])
 
             if self.new_containerapp["properties"]["template"]["serviceBinds"] is None:
                 self.new_containerapp["properties"]["template"]["serviceBinds"] = []
 
-            for item in self.new_containerapp["properties"]["template"]["serviceBinds"]:
-                for update_item in service_bindings_def_list:
-                    if update_item["name"] in item.values():
-                        item["serviceId"] = update_item["serviceId"]
-                        service_bindings_used_map[update_item["name"]] = True
-
             for update_item in service_bindings_def_list:
```

Comment:
This is for bind dev service update existing binding service, currently the command can be executed repeatedly.
for example: 
`az containerapp update -n xy -g arc-appint-forxinyu12-rg0 --bind redis`

But in your pr, it will throw following error:
```
File "C:\Users\xinyupang.REDMOND\source\azure-cli-extensions\src\containerapp\azext_containerapp\containerapp_decorator.py", line 867, in construct_payload
    self.set_up_service_bindings()
  File "C:\Users\xinyupang.REDMOND\source\azure-cli-extensions\src\containerapp\azext_containerapp\containerapp_decorator.py", line 986, in set_up_service_bindings
    check_bindings_and_raise_error(self.cmd, service_connectors_def_list, service_bindings_def_list, self.get_argument_resource_group_name(), self.get_argument_name())
  File "C:\Users\xinyupang.REDMOND\source\azure-cli-extensions\src\containerapp\azext_containerapp\_utils.py", line 133, in check_bindings_and_raise_error
    raise ValidationError(
azure.cli.core.azclierror.ValidationError: Binding names across managed and dev services should be unique.

cli.azure.cli.core.azclierror: Binding names across managed and dev services should be unique.
```

Reply:
I think it is breaking change and not expected.

Reply:
Hi @bgashirabake 
I think this is not relate to the [containerapp] az containerapp create --bind kafkaconfluent: Add support for Managed Kafka], this is for dev service, I recommend not change it, if you really want to improve it, please open another PR.
BTW, command for `update --bind` should be able to execut repeatedly.

Reply:
Hi @bgashirabake , you can ask @yash-nisar for more information about the logic for dev service here.